src/pygama/lgdo/table.py

```diff
@@ -205,3 +205,63 @@ def get_dataframe(self, cols: list[str] = None, copy: bool = False) -> pd.DataFr
                 df[col] = self[col].nda
 
         return df
+
+    def eval(self, expr_config: dict) -> Table:
+        """Apply column operations to the table and return a new table holding
+        the resulting columns.
+
+        Currently defers all the job to :meth:`pandas.DataFrame.eval`. This
+        might change in the future.
+
+        Parameters
+        ----------
+        expr_config
+            dictionary that configures expressions according the following
+            specification:
+
+            .. code-block:: js
+
+                {
+                    "O1": {
+                        "expression": "@p1 + @p2 * a**2",
+                        "parameters": {
+                            "p1": "2",
+                            "p2": "3"
+                        }
+                    },
+                    "O2": {
+                        "expression": "O1 - b"
+                    }
+                    // ...
+                }
+
+            where:
+
+            - ``expression`` is an expression string supported by
+              :meth:`pandas.DataFrame.eval` (see also `here
+              <https://pandas.pydata.org/pandas-docs/stable/user_guide/enhancingperf.html#expression-evaluation-via-eval>`_
+              for documentation).
+            - ``parameters`` is a dictionary of function parameters. Passed to
+              :meth:`pandas.DataFrame.eval` as `local_dict` argument.
+
+
+        Warning
+        -------
+        Blocks in `expr_config` must be ordered according to mutual dependency.
+        """
+        # don't want to modify input
+        df = self.get_dataframe().copy()
```

Comment:
is the copy really necessary? if it is, we can put `copy=True` in the arg of `get_dataframe()` (avoids making 2 dataframes). But I think we want to avoid reallocation of the data arrays, and just have the df use the ones in the table. They should not get modified by eval. See also below.

Reply:
That is a shallow copy right? Means that only keys and references are actually copied. If this is correct, should be quite inexpensive. I'm copying it because the function would otherwise modify the input table by adding new columns. Maybe we could do some garbage collection at the end of the function and delete the copy?

Reply:
you would need to do `df.copy(deep=False)` to get a shallow copy

If the df is built with `copy=True` then it's just a few of the table columns. When eval adds new columns to the df the lgdo table knows nothing about them. It should remain unmodified.

Reply:
I see what you mean now. I will add the `copy=True` argument to `get_dataframe`.